CMSIS/NN/Source/ConvolutionFunctions/arm_depthwise_conv_s8_opt.c

```diff
@@ -0,0 +1,317 @@
+ *
+ * www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an AS IS BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/* ----------------------------------------------------------------------
+ * Project:      CMSIS NN Library
+ * Title:        arm_depthwise_conv_s8_opt.c
+ * Description:  Optimized s8 depthwise separable convolution function for
+ *               channel multiplier of 1.
+ *
+ * $Date:        August 2019
+ * $Revision:    V.1.0.0
+ *
+ * Target Processor:  Cortex-M cores
+ *
+ * -------------------------------------------------------------------- */
+
+#include "arm_math.h"
+#include "arm_nnsupportfunctions.h"
+#include "arm_nnfunctions.h"
+
+/**
+ *  @ingroup groupNN
+ */
+
+/**
+ * @addtogroup NNConv
+ * @{
+ */
+
+/**
+   * @brief Optimized s8 depthwise convolution function with constraint that in_channel equals out_channel
+   *
+   * @note
+   *        - Refer prototype header file for details.
+   *
+   */
+arm_status arm_depthwise_conv_s8_opt(const q7_t *input,
+                                     const uint16_t input_x,
+                                     const uint16_t input_y,
+                                     const uint16_t input_ch,
+                                     const q7_t *kernel,
+                                     const uint16_t output_ch,
+                                     const uint16_t kernel_x,
+                                     const uint16_t kernel_y,
+                                     const uint16_t pad_x,
+                                     const uint16_t pad_y,
+                                     const uint16_t stride_x,
+                                     const uint16_t stride_y,
+                                     const q7_t *bias,
+                                     q7_t *output,
+                                     const int32_t *output_shift,
+                                     const int32_t *output_mult,
+                                     const uint16_t output_x,
+                                     const uint16_t output_y,
+                                     const int32_t output_offset,
+                                     const int32_t input_offset,
+                                     const int32_t output_activation_min,
+                                     const int32_t output_activation_max,
+                                     const uint16_t dilation_x,
+                                     const uint16_t dilation_y,
+                                     q15_t *buffer_a)
+{
+
+
+    /* Check input constraints input_ch == output_ch */
+    if (input_ch != output_ch)
+    {
+        return ARM_MATH_SIZE_MISMATCH;
+    }
+
+// #if defined(ARM_MATH_LOOPUNROLL) && defined(ARM_MATH_DSP)
```

Comment:
It should be uncommented and the #if 1 below removed.

Reply:
ah..my bad. I'll fix it